### PR TITLE
[BugFix] set _column_pool_memory_tracker be shared_ptr for gracefully exit BE (backport #17687)

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
cp https://github.com/StarRocks/starrocks/pull/13427 and https://github.com/StarRocks/starrocks/pull/17687

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by sduzh — DISMISSED** [review]

**Review by Seaven — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by imay** [comment]
run starrocks_be_unittest

File: be/src/common/logconfig.cpp [hunk 1]
```diff
@@ -112,6 +125,13 @@ bool init_glog(const char* basename, bool install_signal_handler) {
         google::InstallFailureSignalHandler();
     }
 
+    // dump trace info may access some runtime stats
+    // if runtime stats broken we won't dump stack
+    if (config::dump_trace_info) {
+        google::InstallFailureWriter(failure_writer);
+        google::InstallFailureFunction(failure_function);
+    }
+
```
Comment (stdpain) [inline be/src/common/logconfig.cpp:134]: remove these code. dup with line 219

Comment (fzhedu) [reply be/src/common/logconfig.cpp]: done

**Review by stdpain — APPROVED** [review]

**Comment by fzhedu** [comment]
run starrocks_be_unittest

**Comment by fzhedu** [comment]
run starrocks_clang-format

**Pull request merged**